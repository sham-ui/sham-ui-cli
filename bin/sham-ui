#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .parse( process.argv );