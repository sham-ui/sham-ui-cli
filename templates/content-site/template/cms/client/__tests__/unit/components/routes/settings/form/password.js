import * as directives from 'sham-ui-directives';
// eslint-disable-next-line max-len
import RoutesSettingsFormPassword  from '../../../../../../src/components/routes/settings/form/password.sfc';
import renderer from 'sham-ui-test-helpers';
import { createDI } from 'sham-ui';

it( 'renders correctly', () => {
    const DI = createDI();
    const meta = renderer( RoutesSettingsFormPassword, {}, {
        DI,
        directives
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'display errors', async() => {
    expect.assertions( 2 );

    const DI = createDI();
    const updateMock = jest.fn();
    DI.bind( 'store', {
        updateMemberPassword: updateMock.mockReturnValueOnce( Promise.reject( {} ) )
    } );

    const meta = renderer( RoutesSettingsFormPassword, {}, {
        DI,
        directives
    } );

    const formData = {
        pass1: 'admin1@gmail.com',
        pass2: 'admin1@gmail.com'
    };
    const { ctx } = meta;
    ctx.container.querySelector( '[name="pass1"]' ).value = formData.pass1;
    ctx.container.querySelector( '[name="pass2"]' ).value = formData.pass2;
    ctx.container.querySelector( '[type="submit"]' ).click();
    ctx.container.querySelector( '[data-test-modal] [data-test-ok-button]' ).click();

    await new Promise( resolve => setImmediate( resolve ) );

    expect( updateMock.mock.calls ).toHaveLength( 1 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
