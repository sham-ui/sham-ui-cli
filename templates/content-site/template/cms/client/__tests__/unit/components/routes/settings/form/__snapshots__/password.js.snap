// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display errors 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "onUpdateSuccess": [Function],
  },
  "html": "
<form class=\\"form-layout\\">
  <!--0-->
  <ul class=\\"errors-panel\\">
    <li>Update password fail</li>
  </ul>
  <!--0-->
  <!--1-->
  <div class=\\"form-group\\">
    <label>Enter your password:</label>
    <input name=\\"pass1\\" type=\\"password\\" autofocus=\\"\\">
  </div>
  <div class=\\"form-group\\">
    <label>Confirm your password:</label>
    <input name=\\"pass2\\" type=\\"password\\">
  </div>
  <!--0-->
  <!--2-->
  <button type=\\"submit\\">
    <!--3-->Update
    <!--4-->
  </button>
  <!--5-->
</form>
<!--0-->
<!--1-->
<!--0-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "onUpdateSuccess": [Function],
  },
  "html": "
<form class=\\"form-layout\\">
  <!--0-->
  <!--0-->
  <!--1-->
  <div class=\\"form-group\\">
    <label>Enter your password:</label>
    <input name=\\"pass1\\" type=\\"password\\" autofocus=\\"\\">
  </div>
  <div class=\\"form-group\\">
    <label>Confirm your password:</label>
    <input name=\\"pass2\\" type=\\"password\\">
  </div>
  <!--0-->
  <!--2-->
  <button type=\\"submit\\">
    <!--3-->Update
    <!--4-->
  </button>
  <!--5-->
</form>
<!--0-->
<!--1-->
<!--0-->
",
}
`;
